source/audio.py

```diff
@@ -0,0 +1,204 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2021 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import atexit
+from comInterfaces.coreAudio.constants import (
+	CLSID_MMDeviceEnumerator,
+	EDataFlow,
+	ERole,
+)
+import comInterfaces.coreAudio.audioclient as audioclient
+import comInterfaces.coreAudio.audiopolicy as audiopolicy
+import comInterfaces.coreAudio.mmdeviceapi as mmdeviceapi
+import comtypes
+import config
+from enum import IntEnum, unique
+import globalVars
+import json
+from logHandler import log
+import nvwave
+from typing import Callable
+import ui
+from utils.displayString import DisplayStringIntEnum
+import _ctypes
+
+VolumeTupleT = tuple[float, float]
+
+
+@unique
+class SoundSplitState(DisplayStringIntEnum):
+	OFF = 0
+	NVDA_LEFT_APPS_RIGHT = 1
+	NVDA_LEFT_APPS_BOTH = 2
+	NVDA_RIGHT_APPS_LEFT = 3
+	NVDA_RIGHT_APPS_BOTH = 4
+	NVDA_BOTH_APPS_LEFT = 5
+	NVDA_BOTH_APPS_RIGHT = 6
+
+	@property
+	def _displayStringLabels(self) -> dict[IntEnum, str]:
+		return {
+			# Translators: Sound split state
+			SoundSplitState.OFF: _("Disabled sound split"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_LEFT_APPS_RIGHT: _("NVDA on the left and applications on the right"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_LEFT_APPS_BOTH: _("NVDA on the left and applications in both channels"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_RIGHT_APPS_LEFT: _("NVDA on the right and applications on the left"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_RIGHT_APPS_BOTH: _("NVDA on the right and applications in both channels"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_BOTH_APPS_LEFT: _("NVDA in both channels and applications on the left"),
+			# Translators: Sound split state
+			SoundSplitState.NVDA_BOTH_APPS_RIGHT: _("NVDA in both channels and applications on the right"),
+		}
+	
+	def getAppVolume(self) -> VolumeTupleT:
+		if self == SoundSplitState.OFF or 'APPS_BOTH' in self.name:
+			return (1.0, 1.0)
+		elif 'APPS_LEFT' in self.name:

```

Comment:
Having two options no longer requires you to do this kind of slightly hacking things.

Reply:
Still, I think i prefer having this stated explicitly. Also, please consider using a match statement, which in a Python 3.10+ word is more readable then a bunch of f/elif statements.

Reply:
I'd also like to see this "hack" replaced by something more maintainable, e.g. in case we change the names or add new members to the enum in the future:

```suggestion
		elif self in [SoundSplitState.NVDA_RIGHT_APPS_LEFT, SoundSplitState.NVDA_BOTH_APPS_LEFT]:
```